client/my-sites/plans-features-main/components/plan-notices.tsx

```diff
@@ -0,0 +1,101 @@
+import Notice from 'calypso/components/notice';
+import { getDiscountByName } from 'calypso/lib/discounts';
+import { usePlanUpgradeCreditsDisplay } from 'calypso/my-sites/plans-features-main/hooks/use-plan-upgrade-credits-display';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import { isCurrentUserCurrentPlanOwner } from 'calypso/state/sites/plans/selectors';
+import { isCurrentPlanPaid } from 'calypso/state/sites/selectors';
+
+export default function PlanNotice( {
+	siteId,
+	isInSignup,
+	visiblePlanNames = [],
+	discountInformation: { withDiscount, discountEndDate },
+}: {
+	visiblePlanNames: PlanSlug[];
+	isInSignup: boolean;
+	siteId: number;
+	discountInformation: {
+		withDiscount: string;
+		discountEndDate: Date;
+	};
+} ) {
+	const translate = useTranslate();
+	const [ isNoticeDismissed, setIsNoticeDismissed ] = useState( false );
+	const currencyCode = useSelector( ( state ) => getCurrentUserCurrencyCode( state ) );
+	const canUserPurchasePlan = useSelector(
+		( state ) =>
+			! isCurrentPlanPaid( state, siteId ) || isCurrentUserCurrentPlanOwner( state, siteId )
+	);
+	const activeDiscount = getDiscountByName( withDiscount, discountEndDate );
+	const { creditsValue, isPlanUpgradeCreditEligible } = usePlanUpgradeCreditsDisplay(
+		siteId,
+		visiblePlanNames
+	);
+
+	const handleDismissNotice = () => setIsNoticeDismissed( true );
+
+	if ( isNoticeDismissed ) {
+		return null;
+	} else if ( ! canUserPurchasePlan ) {
+		return (
+			<Notice
+				className="plan-features__notice"
+				showDismiss={ true }
+				onDismissClick={ handleDismissNotice }
+				status="is-info"
+			>
+				{ translate(
+					'This plan was purchased by a different WordPress.com account. To manage this plan, log in to that account or contact the account owner.'
+				) }
+			</Notice>
+		);
+	} else if ( activeDiscount ) {
+		return (
+			<Notice
+				className="plan-features__notice-credits"
+				showDismiss={ true }
+				onDismissClick={ handleDismissNotice }
+				icon="info-outline"
+				status="is-success"
+			>
+				{ activeDiscount?.plansPageNoticeTextTitle && (
+					<strong>
+						{ activeDiscount?.plansPageNoticeTextTitle }
+						<br />
+					</strong>
+				) }
+				{ activeDiscount.plansPageNoticeText }
+			</Notice>
+		);
+	} else if ( isPlanUpgradeCreditEligible && ! isInSignup ) {
+		return (
+			<Notice
+				className="plan-features__notice-credits"
+				showDismiss={ true }
+				onDismissClick={ handleDismissNotice }
+				icon="info-outline"
+				status="is-success"
+			>
+				{ translate(
+					'You have {{b}}%(amountInCurrency)s{{/b}} of pro-rated credits available from your current plan. ' +
+						'Apply those credits towards an upgrade before they expire!',
+					{
+						args: {
+							amountInCurrency: formatCurrency( creditsValue, currencyCode ?? '' ),
+						},
+						components: {
+							b: <strong />,
+						},
+					}
+				) }
+			</Notice>
+		);
+	}
```

Comment:
We probably don't need the `else` here since all of these are return statements. So this could be something like the following for a small readability win.

```js
if ( isNoticeDismissed ) {
  return null;
}

if ( ! canUserPurchasePlan ) {
  return <Notice...
}
```

Reply:
I actually wanted to explicitly highlight the fact that this is a linear chain of conditions. Do you think it's more beneficial to **not** have the `else` 🤔 ?

Reply:
Since we return early from each statement, it would still be a linear chain. TBH, there's not really much of a difference here, and I highlighted it here just in case you had not considered this approach. Either way works! 🙂